File: 프로그래머스/0/120923. 연속된 수의 합/연속된 수의 합.js
function solution(num, total) {
    function getFirstNum(num, total) {
        // num이 짝수인 경우
        if(num % 2 === 0){
            return Math.ceil(total / num) - (num / 2);
        }
        // num이 홀수인 경우
        else{
            return total / num - Math.floor(num / 2);
        }
    }
    let firstNum = getFirstNum(num,total);
    let answer = Array(num).fill(firstNum).map((v,idx)=>v+idx);
    return answer;
}

File: 프로그래머스/0/120923. 연속된 수의 합/연속된 수의 합.java
class Solution {
    public int[] solution(int num, int total) {
        int[] answer = new int[num];
        int temp = 0;
        for(int i=0;i<num;i++){
            temp+=i;
        }
        int value = (total-temp)/num;

        for(int i=0;i<num;i++){
            answer[i]=i+value;
        }

        return answer;
    }
}